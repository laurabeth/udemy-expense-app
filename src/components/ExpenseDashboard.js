import React from "react";

const ExpenseDashboard = () => <div> This is the dashboard component </div>;

export default ExpenseDashboard;
