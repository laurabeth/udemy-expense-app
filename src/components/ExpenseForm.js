import React from "react";
import moment from "moment";
import "react-dates/initialize";
import { SingleDatePicker } from "react-dates";
import "react-dates/lib/css/_datepicker.css";

const { useState } = React;

const ExpenseForm = () => {
  const [description, setDescription] = useState({ value: "" });
  const [amount, setAmount] = useState({ value: 0 });
  const [note, setNote] = useState({ value: "" });
  const [createdOn, setCreatedOn] = useState(moment());
  const [calendarFocused, setCalendarFocused] = useState(false);

  const onDescriptionChange = (event) => {
    setDescription({ value: event.target.value });
  };

  const onAmountChange = (event) => {
    const input = event.target.value;
    if (!amount || input.match(/^\d{1,}(\.\d{0,2})?$/)) {
      setAmount({ value: input });
    }
  };

  const onDateChange = (date) => {
    if (date) {
      setCreatedOn(date);
    }
  };

  const onNoteChange = (event) => {
    setNote({ value: event.target.value });
  };

  return (
    <div>
      <form>
        <input
          onChange={onDescriptionChange}
          placeholder="Description"
          type="text"
          value={description.value}
        />
        <input
          onChange={onAmountChange}
          placeholder="Amount"
          type="string"
          value={amount.value}
        />
        <SingleDatePicker
          date={createdOn}
          focused={calendarFocused}
          isOutsideRange={() => false}
          numberOfMonths={1}
          onDateChange={onDateChange}
          onFocusChange={({ focused }) => setCalendarFocused(focused)}
        />
        <textarea
          onChange={onNoteChange}
          placeholder="Add a note for your expense (optional)"
          value={note.value}
        ></textarea>
        <button>Add Expense</button>
      </form>
    </div>
  );
};

export default ExpenseForm;
