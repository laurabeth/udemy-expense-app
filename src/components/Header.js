import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
  <header>
    <h1>Expenses</h1>
    <NavLink activeClassName="is-active" exact to="/">
      Dashboard
    </NavLink>
    <span> | </span>
    <NavLink activeClassName="is-active" to="/create">
      Create Expense
    </NavLink>
    <span> | </span>
  </header>
);

export default Header;
