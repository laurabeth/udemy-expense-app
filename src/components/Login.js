import React from "react";
import { connect } from "react-redux";
import { authenticateAsync } from "../actions/auth";

const Login = ({ authenticateAsync }) => {
  return (
    <div>
      <button onClick={authenticateAsync}>Log in</button>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  authenticateAsync: () => dispatch(authenticateAsync()),
});

export default connect(undefined, mapDispatchToProps)(Login);
