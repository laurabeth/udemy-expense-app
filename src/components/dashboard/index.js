export { default as ExpenseDashboard } from "./ExpenseDashboard";
export { default as ExpenseList } from "./ExpenseList";
export { default as ExpenseListFilters } from "./ExpenseListFilters";
export { default as ExpenseListItem } from "./ExpenseListItem";
