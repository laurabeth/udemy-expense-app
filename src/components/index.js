export * from "./dashboard";
export { default as AddExpense } from "./AddExpense";
export { default as EditExpense } from "./EditExpense";
export { default as ExpenseForm } from "./ExpenseForm";
export { default as Header } from "./Header";
export { default as Help } from "./Help";
export { default as NotFound } from "./NotFound";
