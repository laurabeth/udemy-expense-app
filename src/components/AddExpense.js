import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { addExpense } from "../actions/expenses";

export const AddExpense = ({ history, onSubmit }) => {
  const submitExpense = (expense) => {
    onSubmit(expense);
    history.push("/");
  };

  return (
    <>
      <h1>AddExpense</h1>
      <ExpenseForm submitExpense={submitExpense} />
    </>
  );
};

const mapDispatchToProps = (dispatch) => ({
  onSubmit: (expense) => dispatch(addExpense(expense)),
});

export default connect(undefined, mapDispatchToProps)(AddExpense);
