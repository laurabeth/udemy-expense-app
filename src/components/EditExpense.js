import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpenseAsync, removeExpenseAsync } from "../actions";

export const EditExpense = ({
  editExpenseAsync,
  expense,
  history,
  removeExpenseAsync,
}) => {
  const submitEdit = (edits) => {
    editExpenseAsync(expense.id, edits);
    history.push("/");
  };
  const handleRemoveExpense = () => {
    removeExpenseAsync(expense.id);
    history.push("/");
  };
  return (
    <div>
      <ExpenseForm expense={expense} submitExpense={submitEdit} />
      <button onClick={handleRemoveExpense}>Remove</button>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),
  };
};

const mapDispatchToProps = (dispatch) => ({
  editExpenseAsync: (id, edits) => dispatch(editExpenseAsync(id, edits)),
  removeExpenseAsync: (id) => dispatch(removeExpenseAsync({ id })),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);
