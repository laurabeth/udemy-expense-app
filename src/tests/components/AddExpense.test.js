import React from "react";
import { shallow } from "enzyme";
import { AddExpense } from "../../components/AddExpense";
import expensesMock from "../fixtures/expenses";

/* eslint-disable no-undef */
describe("add expense", () => {
  it("renders", () => {
    const onSubmit = jest.fn();
    const history = { push: jest.fn() };
    const wrapper = shallow(<AddExpense history={history} onSubmit={onSubmit} />);
    expect(wrapper).toMatchSnapshot();
  });
});
