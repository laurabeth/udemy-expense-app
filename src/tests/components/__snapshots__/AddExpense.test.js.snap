// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add expense renders 1`] = `
<Fragment>
  <h1>
    AddExpense
  </h1>
  <ExpenseForm
    submitExpense={[Function]}
  />
</Fragment>
`;
